test/com/tfl/billing/JourneyCostCalculatorTest.java

```diff
@@ -0,0 +1,130 @@
+        peakTime = new Date();
+        offPeakTime = new Date();
+        c = CustomerDatabase.getInstance().getCustomers().get(0);
+        c1 = CustomerDatabase.getInstance().getCustomers().get(1);
+        c2 = CustomerDatabase.getInstance().getCustomers().get(2);
+        eventLog = Arrays.asList(new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyStart(c1.cardId(), OysterReaderLocator.atStation(Station.HOLBORN).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id()), new JourneyEnd(c1.cardId(), OysterReaderLocator.atStation(Station.CHANCERY_LANE).id()), new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.OXFORD_CIRCUS).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id()), new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.HOLBORN).id()));
+        customerJourneys = Arrays.asList(new Journey(new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id())));
+        costCalculator = new JourneyCostCalculator();
+    }
+
+    private long hoursToMillis(int hour) {
+        return hour*60*60*1000;
+    }
+
+    @Test
+    public void getAllCardScansForCustomer()  {
+        List<JourneyEvent> expected = Arrays.asList(new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id()), new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.OXFORD_CIRCUS).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id()), new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.HOLBORN).id()));
+        List<JourneyEvent> testResult = costCalculator.getJourneyEvents(c,eventLog);
+
+
+        //assert by value using flag
+        boolean flag = true;
+        for (int i = 0; i < testResult.size(); i++) {
+            if(testResult.get(i).readerId() != expected.get(i).readerId() || testResult.get(i).cardId() != expected.get(i).cardId()) {
+                flag = false;
+            }
+        }
+        assertTrue(flag);
+    }
+
+    @Test
+    public void getAllCompletedJourneys()  {
+        List<Journey> expected = Arrays.asList(new Journey(new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEnd(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id())), new Journey( new JourneyStart(c1.cardId(), OysterReaderLocator.atStation(Station.HOLBORN).id()), new JourneyEnd(c1.cardId(), OysterReaderLocator.atStation(Station.CHANCERY_LANE).id())), new Journey(new JourneyStart(c.cardId(), OysterReaderLocator.atStation(Station.OXFORD_CIRCUS).id()), new JourneyEnd(c.cardId(),  OysterReaderLocator.atStation(Station.VICTORIA_STATION).id())));
```

Comment:
This looks bad and is bad, sorry. It takes 10 min to understand what this is. Can't we have a helper method which would generate this array for you?